src/CMakeLists.txt

```diff
@@ -317,11 +317,15 @@ endif()
 
 add_dependencies(ncnn generate-spirv)
 
-if(NCNN_OPENMP AND OpenMP_CXX_FOUND)
+if(NCNN_OPENMP AND (OpenMP_CXX_FOUND OR OPENMP_FOUND))
     if(NCNN_CMAKE_VERBOSE)
         message("Building with OpenMP")
     endif()
-    target_link_libraries(ncnn PUBLIC OpenMP::OpenMP_CXX)
+    if(OpenMP_CXX_FOUND)
+        target_link_libraries(ncnn PUBLIC OpenMP::OpenMP_CXX)
+    else()
+        target_link_libraries(ncnn PRIVATE "${OpenMP_CXX_FLAGS}")
```

Comment:
${OpenMP_CXX_FLAGS} is a compile flag, I don't think it is correct to link with it


For old cmake version, you may refer
https://stackoverflow.com/questions/12399422/how-to-set-linker-flags-for-openmp-in-cmakes-try-compile-function

Reply:
For gcc on linux, both `CFLAGS` and `LDFLAGS` need `-fopenmp` flag. But the `OpenMP_EXE_LINKER_FLAGS` or any other openmp ldflag doesn't exist. So I used `OpenMP_CXX_FLAGS` instead of `OpenMP_EXE_LINKER_FLAGS` here, even it is a compile flag. Maybe it is not a good idea, but I didn't find a better way now.